File: java/exercises/02-intermediate/ex068_junit_parameterized/PrimeCheckerTest.java
package fewolearning.exercises.intermediate.ex068_junit_parameterized;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.junit.jupiter.api.Assertions.assertEquals;

class PrimeCheckerTest {

    @ParameterizedTest
    @CsvSource({
            "-7, false",
            "0, false",
            "1, false",
            "2, true",
            "3, true",
            "4, false",
            "17, true",
            "18, false",
            "97, true",
            "100, false"
    })
    void identifiesPrimalityCorrectly(int candidate, boolean expected) {
        assertEquals(expected, PrimeChecker.isPrime(candidate));
    }
}

File: kotlin/exercises/02-intermediate/ex060_junit_parameterized/PrimeCheckerTest.kt
package fewolearning.exercises.intermediate.ex060_junit_parameterized

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

class PrimeCheckerTest {

    @ParameterizedTest
    @ValueSource(ints = [2, 3, 5, 7, 11, 13, 97])
    fun isPrimeRecognizesPrimeNumbers(candidate: Int) {
        assertTrue(isPrime(candidate))
    }

    @ParameterizedTest
    @ValueSource(ints = [0, 1, 4, 6, 8, 9, 100])
    fun isPrimeRejectsNonPrimeNumbers(candidate: Int) {
        assertFalse(isPrime(candidate))
    }

    @ParameterizedTest
    @ValueSource(ints = [-5, -1])
    fun isPrimeRejectsNegativeNumbers(candidate: Int) {
        assertFalse(isPrime(candidate))
    }
}
